File: src/my_project/easy_problems/from151To200/uncommon_words_from_two_sentences.py
from typing import List 

class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:

        s1 = s1.split(' ')
        s2 = s2.split(' ')
        dictionary_one = dict()
        dictionary_two = dict() 

        for word in s1: 
            dictionary_one[word] = 0

        for word in s2: 
            dictionary_two[word] = 0

        for word in s1: 
            dictionary_one[word] += 1
        for word in s2: 
            dictionary_two[word] += 1
        solution_list = list()

        first_list = list()
        second_list = list()

        for word in s1:
            if word not in s2 and dictionary_one[word] == 1:
                first_list.append(word)

        for word in s2:
            if word not in s1 and word not in first_list and dictionary_two[word] ==1:
                second_list.append(word)

        for word in first_list + second_list:
            solution_list.append(word)

        return solution_list


solution = Solution()

print(solution.uncommonFromSentences(s1 = "this apple is sweet", s2 = "this apple is sour"))

print(solution.uncommonFromSentences(s1 = "apple apple", s2 = "banana"))

File: src/my_project_typescript/easy_problems_typescript/from51To100/uncommonWordsFromTwoSentences.ts
function uncommonFromSentences(s1: string, s2: string): string[] {

    let s1Array: Array<string> = []
    let s2Array: Array<string> = []

    s1Array = s1.split(' ')
    s2Array = s2.split(' ')

    let dictionaryOne: any = {}
    let dictionaryTwo: any = {}

    s1Array.map( word => {
        dictionaryOne[word] = 0
    })

    s2Array.map(word => {
        dictionaryTwo[word] = 0
    })

    s1Array.map(word => {
        dictionaryOne[word]+=1
    })

    s2Array.map(word => [
        dictionaryTwo[word] += 1
    ])

    let solutionList: Array<string> = []
    let firstList: Array<string> = []
    let secondList: Array<string> = []

    s1Array.map(word => {
        if (!s2Array.includes(word) && dictionaryOne[word] ==1 ){
            firstList.push(word)
        }
    })

    s2Array.map(word => {
        if (!s1Array.includes(word) && !firstList.includes(word) && dictionaryTwo[word] ==1){
            secondList.push(word)
        }
    })

    firstList.concat(secondList).map(word => {
        solutionList.push(word)
    })

    return solutionList
};

console.log(uncommonFromSentences("this apple is sweet",  "this apple is sour"))

console.log(uncommonFromSentences("apple apple",  "banana"))
